### PR TITLE
feat: dynamic assistance based on user question

### PR DESCRIPTION
**Pull request by Harish-osmosys**
Soochna :https://soochna.osmosys.co/books/research-development/page/dynamic-assistance-picking

Sample Converstion log: 
[sample_conversation.txt](https://github.com/OsmosysSoftware/dynamic-assistant-poc/files/14670366/sample_conversation.txt)

## Summary by CodeRabbit

- **New Features**
    - Introduced a dynamic AI assistant feature that interacts with OpenAI's API, capable of running two different assistants (policy or skill) based on user input.

- **Documentation**
    - Added setup instructions for the dynamic-assistant proof of concept, including prerequisites, environment setup, and running the application.

- **Chores**
    - Added a `.env.example` file for API key storage.
    - Updated `.gitignore` to exclude the `assistant/` virtual environment directory.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            time.sleep(5)
+
+def handle_user_query():
+    while True:
+        user_query = input("Please enter your question (type 'exit' to quit): ")
+
+        if user_query.lower() == "exit":
+            break
+
+        if "skill" in user_query.lower():
```
Comment (osm-vishnukyatannawar) [inline app.py]: I don't think we can rely on this. What happens if I ask question like

```
Who are our .NET resources?
How many hours do I have to work at office?
```

Just checking for skill and policy is not enough. We needed more intelligence here. Why not keep an assistant to ask that? which we already discussed over call.

**Review by osm-vishnukyatannawar — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (Harish-osmosys) [reply app.py]: Updated the classification logic 

image

File: README.md [hunk 2]
```diff
@@ -1,2 +1,38 @@
 # dynamic-assistant-poc
+This is a POC to demonstrate how to dynamically pick an assistant to route queries to the related assistant.
 This is a POC to demonstrate how to dynamic pick assistant to route queries to related assistant.
+
+
+## Prerequisites
+
+Make sure you have python(3.11.7) and pip(24.0) installed
```
Comment (Harish-osmosys) [inline README.md:7]: I specified the currently used version and these are the latest version only no need to explictly say  about higher version

File: README.md [hunk 3]
```diff
@@ -1,2 +1,38 @@
...
+```bash
+python --version
+pip --version
+```
+
+If Python or pip is not installed, please visit [Python's official website](https://www.python.org/downloads/) for installation instructions.
+
+## Setup
+
+Update the .env file with the OpenAI_Server_Key. You can obtain this key from your OpenAI account dashboard
```
Comment (Harish-osmosys) [inline README.md:20]: the key is present in the .env file so we are safe as .env file is added in gitignore

File: README.md [hunk 4]
```diff
@@ -1,2 +1,38 @@
...
+
+# If you are using the fish shell use this to activate virtual environment
+source assistant/bin/activate.fish
+
+# If you are using the bash shell use this to activate virtual environment
+source assistant/bin/activate
+
+pip install --upgrade openai
+
+python app.py
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply README.md]: We don't need requiremetns.txt file as we need only one package here , we can specify it here only

File: README.md [hunk 5]
```diff
@@ -1,2 +1,38 @@
...
+```bash
+python --version
+pip --version
+```
+
+If Python or pip is not installed, please visit [Python's official website](https://www.python.org/downloads/) for installation instructions.
+
+## Setup
+
+Update the .env file with the OpenAI_Server_Key. You can obtain this key from your OpenAI account dashboard
```
Comment (Harish-osmosys) [inline README.md:20]: the key is present in the .env file so we are safe as .env file is added in gitignore

File: README.md [hunk 6]
```diff
@@ -1,2 +1,38 @@
 # dynamic-assistant-poc
+This is a POC to demonstrate how to pick an assistant dynamically to route queries to the related assistant.
 This is a POC to demonstrate how to dynamic pick assistant to route queries to related assistant.
+
+
+## Prerequisites
+
+Make sure you have python(3.11.7) and pip(24.0) installed
```
Comment (Harish-osmosys) [inline README.md:7]: I specified the currently used version and these are the latest version only no need to explictly say about higher version

File: README.md [hunk 7]
```diff
@@ -1,2 +1,37 @@
...
+
+python3 -m venv assistant
+
+# If you are using the fish shell use this to activate virtual environment
+source assistant/bin/activate.fish
+
+# If you are using the bash shell use this to activate virtual environment
+source assistant/bin/activate
+
+pip install --upgrade openai python-dotenv
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply README.md:34]: We don't need requiremetns.txt file as we need only two package here , we can specify it here only later on if we have more packages than we can do it

File: README.md [hunk 8]
```diff
@@ -1,2 +1,37 @@
...
+```bash
+python --version
+pip --version
+```
+
+If Python or pip is not installed, please visit [Python's official website](https://www.python.org/downloads/) for installation instructions.
+
+## Setup
+
+Update the .env file with the OpenAI_Server_Key. You can obtain this key from your OpenAI account dashboard
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply README.md:20]: the key is present in the .env file so we are safe as .env file is added in gitignore

File: README.md [hunk 9]
```diff
@@ -1,2 +1,37 @@
 # dynamic-assistant-poc
 This is a POC to demonstrate how to dynamic pick assistant to route queries to related assistant.
+
+
+## Prerequisites
+
+Make sure you have python(3.11.7) and pip(24.0) installed
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply README.md:7]: I specified the currently used version and these are the latest version only no need to explictly say about higher version

File: README.md [hunk 10]
```diff
@@ -1,2 +1,37 @@
 # dynamic-assistant-poc
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply README.md:1]: Updated in https://github.com/OsmosysSoftware/dynamic-assistant-poc/pull/2/commits/1a514f87c6ce335522b672b7e6f1fb3b8b12662f

**Review by osm-vishnukyatannawar — APPROVED** [review]

**Comment by osm-vishnukyatannawar** [comment]
This is POC code.

For now I will merge it without having to check if the code follows standards.
We just wanted to see the working if it all we can have dynamic assistant.

**Pull request merged**